lib/rules/prefer-find-by.ts

```diff
@@ -111,6 +111,226 @@ export default createTestingLibraryRule<Options, MessageIds>({
+    ) {
+      if (
+        !isCallExpression(node.body) ||
+        !isMemberExpression(node.body.callee)
+      ) {
+        return null;
+      }
+
+      // expect(getByText).toBeInTheDocument() shape
+      if (
+        isCallExpression(node.body.callee.object) &&
+        isCallExpression(node.body.callee.object.arguments[0]) &&
+        ASTUtils.isIdentifier(node.body.callee.object.arguments[0].callee)
+      ) {
+        return node.body.callee.object.arguments[0].callee.name;
+      }
+
+      if (!ASTUtils.isIdentifier(node.body.callee.property)) {
+        return null;
+      }
+
+      // expect(screen.getByText).toBeInTheDocument() shape
+      if (
+        isCallExpression(node.body.callee.object) &&
+        isCallExpression(node.body.callee.object.arguments[0]) &&
+        isMemberExpression(node.body.callee.object.arguments[0].callee) &&
+        ASTUtils.isIdentifier(
+          node.body.callee.object.arguments[0].callee.property
+        )
+      ) {
+        return node.body.callee.object.arguments[0].callee.property.name;
+      }
+
+      // expect(screen.getByText).not shape
+      if (
+        isMemberExpression(node.body.callee.object) &&
+        isCallExpression(node.body.callee.object.object) &&
+        isCallExpression(node.body.callee.object.object.arguments[0]) &&
+        isMemberExpression(
+          node.body.callee.object.object.arguments[0].callee
+        ) &&
+        ASTUtils.isIdentifier(
+          node.body.callee.object.object.arguments[0].callee.property
+        )
+      ) {
+        return node.body.callee.object.object.arguments[0].callee.property.name;
+      }
+
+      // expect(getByText).not shape
+      if (
+        isMemberExpression(node.body.callee.object) &&
+        isCallExpression(node.body.callee.object.object) &&
+        isCallExpression(node.body.callee.object.object.arguments[0]) &&
+        ASTUtils.isIdentifier(
+          node.body.callee.object.object.arguments[0].callee
+        )
+      ) {
+        return node.body.callee.object.object.arguments[0].callee.name;
+      }
+
+      return node.body.callee.property.name;
+    }
+
+    function getWrongQueryName(node: TSESTree.ArrowFunctionExpression) {
+      if (!isCallExpression(node.body)) {
+        return null;
+      }
+
+      // expect(() => getByText) and expect(() => screen.getByText) shape
+      if (
+        ASTUtils.isIdentifier(node.body.callee) &&
+        helpers.isSyncQuery(node.body.callee)
+      ) {
+        return node.body.callee.name;
+      }
+
+      return getWrongQueryNameInAssertion(node);
+    }
+
+    function getCaller(node: TSESTree.ArrowFunctionExpression) {
+      if (
+        !isCallExpression(node.body) ||
+        !isMemberExpression(node.body.callee)
+      ) {
+        return null;
+      }
+
+      if (ASTUtils.isIdentifier(node.body.callee.object)) {
+        // () => screen.getByText
+        return node.body.callee.object.name;
+      } else if (
```

Comment:
These `else if` can be simplified to just `if` since you are returning inside the statement body right?